### ISSUE TITLE
Nonstandard\Fields implement Rfc4122\FieldsInterface?

### ISSUE THREAD

**Issue by simPod** [ramsey/uuid#317]
I wonder why does `Ramsey\Uuid\Nonstandard\Fields` described as `Nonstandard UUID fields do not conform to the RFC 4122 standard` implement `Ramsey\Uuid\Rfc4122\FieldsInterface` that implies `RFC 4122` compliance.

It seems confusing. Is the design intended?

**Comment by ramsey** [comment]
This design is intended for backwards compatibility. If any UUIDs arise that are non-standard and have different field definitions, then I would create a separate Fields class and interface for them.

**Comment by simPod** [comment]
So there's no point for me trying to find any relation between non-standard and RFC 4122 UUIDs. It's just for lib BC design purposes. If it is so I guess we can close this.

**Comment by ramsey** [comment]
The primary reason is for anyone who might get UUIDs from another source.

As part of 4.0, I increased the strictness of following RFC 4122 for *standard* UUIDs. As a result of this change in strictness, if anyone used `Uuid::fromString()` and the UUIDs were invalid (according to RFC 4122), then the library would throw an exception.

To avoid breaking everyone's apps (since many might have UUIDs that were generated somewhere else), I decided not to throw an exception and, rather, use a different type for these UUIDs (`Nonstandard\Uuid`). However, non-standard UUIDs (unless using some other defined standard for them) might really have intended as regular RFC 4122 UUIDs, but whatever generated them could have put the version/variant bits in the wrong places (or not set them at all). 

As a result, I err on the side of caution here and assume it's supposed to have the same fields as an RFC 4122 UUID.

I hope this makes sense. 😁
